In [6]:
import json 
import pandas as pd
from collections import Counter

with open('retrieval_base_layer7_bert-base-uncased_layer1_p4_k100.json', 'r') as f:
    encodecmae_bert = json.load(f)

with open('retrieval_mel256-ec-base_layer7_bert-base-uncased_layer0_p4_k100.json', 'r') as f:
    encodecmae_mel_bert = json.load(f)

with open('retrieval_wav2vec2_layer6_bert-base-uncased_layer0_p4_k100.json', 'r') as f:
    w2v_bert = json.load(f)

# Incorrectly returned words

We consider words to be incorrectly returned if the word that should appear is not in the top 5.

In [7]:
keys_test = [encodecmae_bert['all_values'][i] for i in encodecmae_bert['rows_deleted']]
total = len(keys_test)
words_encodec = []
count = 0
for i in range(total):
    word = keys_test[i]
    k_1 = encodecmae_bert['retrieval_indices'][i]
    if not any(encodecmae_bert['keys_anchors'][index] == word for index in k_1 if index < len(encodecmae_bert['keys_anchors'])):
        word_retrieval =  [encodecmae_bert['keys_anchors'][k_1] for k_1 in encodecmae_bert['retrieval_indices'][i]]
        words_encodec.append([word, word_retrieval])
        count +=1 
        print(f"word: {word} retrieval:{word_retrieval}, indice {i}")    
        print('---------')
print(f'porcentaje de no devueltas: {count/total}')        
print(f'palabras: {count}') 

word: some retrieval:['insist', 'sooner', 'was', 'miss', 'was'], indice 0
---------
word: this retrieval:['the', 'the', 'the', 'the', 'the'], indice 1
---------
word: fire retrieval:['are', 'are', 'are', 'are', 'are'], indice 3
---------
word: never retrieval:['am', 'am', 'am', 'am', 'am'], indice 5
---------
word: excitement retrieval:['nile', 'wonder', 'autumn', 'general', 'twilight'], indice 9
---------
word: probably retrieval:['wildly', 'emperors', 'little', 'solved', 'little'], indice 10
---------
word: cross retrieval:['karl', 'karl', 'hot', 'crowd', 'casus'], indice 12
---------
word: and retrieval:['in', 'in', 'in', 'in', 'in'], indice 13
---------
word: no retrieval:['mode', 'mode', 'mode', 'mode', 'mode'], indice 15
---------
word: pocket retrieval:['back', 'back', 'up', 'up', 'up'], indice 16
---------
word: voice retrieval:['oil', 'war', 'tours', 'holgrave', 'will'], indice 17
---------
word: an retrieval:['in', 'in', 'in', 'in', 'in'], indice 18
---------
word: keep retri

In [8]:
keys_test = [encodecmae_mel_bert['all_values'][i] for i in encodecmae_mel_bert['rows_deleted']]
total = len(keys_test)
words_encodec_mel = []
count = 0
for i in range(total):
    word = keys_test[i]
    k_1 = encodecmae_mel_bert['retrieval_indices'][i]
    if not any(encodecmae_mel_bert['keys_anchors'][index] == word for index in k_1 if index < len(encodecmae_mel_bert['keys_anchors'])):
        word_retrieval =  [encodecmae_mel_bert['keys_anchors'][k_1] for k_1 in encodecmae_mel_bert['retrieval_indices'][i]]
        words_encodec_mel.append([word, word_retrieval])
        count +=1 
        print(f"word: {word} retrieval:{word_retrieval}, indice {i}")    
        print('---------')
print(f'porcentaje de no devueltas: {count/total}')        
print(f'palabras: {count}') 

word: this retrieval:['was', 'was', 'was', 'was', 'was'], indice 1
---------
word: fire retrieval:['are', 'are', 'are', 'are', 'are'], indice 3
---------
word: excitement retrieval:['it', 'it', 'it', 'it', 'it'], indice 9
---------
word: and retrieval:['in', 'in', 'in', 'in', 'in'], indice 13
---------
word: garden retrieval:['god', 'god', 'god', 'god', 'god'], indice 14
---------
word: no retrieval:['know', 'know', 'know', 'know', 'know'], indice 15
---------
word: pocket retrieval:['talk', 'talk', 'talk', 'talk', 'talk'], indice 16
---------
word: recipe retrieval:['be', 'be', 'be', 'be', 'be'], indice 20
---------
word: that retrieval:['the', 'the', 'the', 'the', 'the'], indice 21
---------
word: their retrieval:['are', 'are', 'are', 'are', 'are'], indice 26
---------
word: turban retrieval:['turn', 'turn', 'turn', 'turn', 'turnips'], indice 29
---------
word: tore retrieval:['wore', 'swore', 'poor', 'poor', 'poor'], indice 31
---------
word: office retrieval:['off', 'off', 'off', '

In [9]:
keys_test = [w2v_bert['all_values'][i] for i in w2v_bert['rows_deleted']]
total = len(keys_test)
words_w2v = []
count = 0
for i in range(total):
    word = keys_test[i]
    k_1 = w2v_bert['retrieval_indices'][i]
    if not any(w2v_bert['keys_anchors'][index] == word for index in k_1 if index < len(w2v_bert['keys_anchors'])):
        word_retrieval =  [w2v_bert['keys_anchors'][k_1] for k_1 in w2v_bert['retrieval_indices'][i]]
        words_w2v.append([word, word_retrieval])
        count +=1 
        print(f"word: {word} retrieval:{word_retrieval}, indice {i}")    
        print('---------')
print(f'porcentaje de no devueltas: {count/total}')              
print(f'palabras: {count}') 

word: cross retrieval:['across', 'across', 'across', 'across', 'across'], indice 12
---------
word: no retrieval:['now', 'now', 'now', 'now', 'now'], indice 15
---------
word: recipe retrieval:['stew', 'christie', 'fish', 'he', 'he'], indice 20
---------
word: wielded retrieval:['sealed', 'will', 'will', 'will', 'will'], indice 34
---------
word: drouet retrieval:['herbert', 'ardour', 'herbert', 'ardor', 'bread'], indice 37
---------
word: panada retrieval:['canada', 'add', 'atlanta', 'add', 'pleroma'], indice 44
---------
word: quarters retrieval:['quarter', 'quarter', 'quarter', 'corner', 'rope'], indice 50
---------
word: olden retrieval:['old', 'old', 'old', 'old', 'old'], indice 54
---------
word: call retrieval:['called', 'called', 'called', 'called', 'called'], indice 55
---------
word: venus retrieval:['v', 'v', 'main', 'main', 'phoebe'], indice 61
---------
word: while retrieval:['wild', 'wild', 'wild', 'wild', 'wild'], indice 63
---------
word: edition retrieval:['tuition', '

How many words are 'incorrectly returned' in each one?

In [11]:
len(keys_test)

7035

In [12]:
len(words_w2v)

1803

In [13]:
len(words_encodec)

4482

In [14]:
len(words_encodec_mel)

3138

In [16]:
words_w2v_counter = [item[0] for item in words_w2v]
words_encodec_counter = [item[0] for item in words_encodec]
words_encodec_mel_counter = [item[0] for item in words_encodec_mel]

counter_w2v = Counter(words_w2v_counter)
counter_encodec = Counter(words_encodec_counter)
counter_encodec_mel = Counter(words_encodec_mel_counter)

common_words = set(words_w2v_counter) & set(words_encodec_counter)  & set(counter_encodec_mel)

print("Words in words_w2v:", len(counter_w2v))
print("Words in words_encodec:", len(counter_encodec))
print("Words in words_encodec:", len(counter_encodec_mel))
print("Common words:", common_words)
print("Number of common words:", len(common_words))

Words in words_w2v: 1523
Words in words_encodec: 2300
Words in words_encodec: 2053
Common words: {'guards', 'sir', 'cannot', 'bad', 'strangeness', 'hi', 'ruined', 'wash', 'ornaments', 'lengths', 'hearing', 'done', 'bow', 'suggestions', 'busy', 'whispered', 'extricating', 'asleep', 'lay', 'unlike', 'regularly', 'attempt', 'advertising', 'descended', 'philosophers', 'divides', 'considered', 'taste', 'knight', 'burst', 'once', 'discretion', 'establishment', 'loose', 'noses', 'maybe', 'holders', 'voices', 'experience', 'boxes', 'woollen', 'operation', 'desolation', 'jar', 'cursing', 'ecclesiastic', 'become', 'hatred', 'morality', 'road', 'fits', 'unmeaning', 'he', 'zoologist', 'looked', 'discern', 'palace', 'newspaper', 'destination', 'boys', 'lacking', 'patriarch', 'gloria', 'trade', 'spoken', 'borne', 'compassionate', 'advancing', 'degrees', 'without', 'crevassed', 'daily', 'toasted', 'determine', 'mouths', 'complain', 'as', 'interested', 'churchwardens', 'swayed', 'reader', 'saddle', 'o

Let's see the words that needed to be returned and how many appearances each one has.

In [18]:
with open('../../words_counts.json', 'r') as json_file:
    words_count = json.load(json_file)

In [19]:
common_word_counts = {word: words_count.get(word, 0) for word in keys_test}
count_one_value = sum(1 for count in common_word_counts.values() if count == 1)

print("Number of words with a count of 1 on key_test:", count_one_value)

Number of words with a count of 1 on key_test: 430


In [20]:
common_word_counts = {word: words_count.get(word, 0) for word in common_words}
count_one_value = sum(1 for count in common_word_counts.values() if count == 1)

print("Number of words with a count of 1 on common_words:", count_one_value)

Number of words with a count of 1 on common_words: 430


There are 430 words that the encoder hadn't seen. What does it return for those words?

In [21]:
count = 0
words_encodec_unique = []
for i in range(len(words_encodec)):
    word = words_encodec[i][0]
    if words_count[word] == 1:
        words_encodec_unique.append(words_encodec[i])
        count += 1
        print(f"word: {word} retrieval:{words_encodec[i][1]}, indice {i}")    

word: wielded retrieval:['little', 'little', 'little', 'little', 'little'], indice 24
word: performance retrieval:['from', 'from', 'from', 'from', 'from'], indice 47
word: charwoman retrieval:['him', 'hour', 'him', 'him', 'him'], indice 52
word: darker retrieval:['our', 'our', 'our', 'our', 'our'], indice 64
word: appointment retrieval:['suppressed', 'reception', 'post', 'bright', 'return'], indice 68
word: woven retrieval:['warm', 'warm', 'warm', 'ball', 'mood'], indice 71
word: bustle retrieval:['up', 'up', 'up', 'up', 'up'], indice 77
word: distinctively retrieval:['ceiling', 'schloss', 'pleasing', 'fully', 'dainty'], indice 93
word: ampler retrieval:['them', 'them', 'them', 'them', 'them'], indice 95
word: pales retrieval:['hoe', 'rows', 'daylight', 'have', 'veil'], indice 106
word: ancestral retrieval:['sunrise', 'known', 'founded', 'man', 'loves'], indice 108
word: administrator retrieval:['admire', 'wagon', 'reeling', 'machine', 'green'], indice 117
word: lantern retrieval:['rei

In [22]:
print(count)

430


In [23]:
count = 0
words_encodec_mel_unique = []
for i in range(len(words_encodec_mel)):
    word = words_encodec_mel[i][0]
    if words_count[word] == 1:
        words_encodec_mel_unique.append(words_encodec_mel[i])
        count += 1
        print(f"word: {word} retrieval:{words_encodec_mel[i][1]}, indice {i}")    

word: wielded retrieval:['will', 'will', 'will', 'will', 'will'], indice 13
word: performance retrieval:['from', 'from', 'from', 'from', 'from'], indice 27
word: charwoman retrieval:['them', 'them', 'them', 'them', 'them'], indice 29
word: darker retrieval:['our', 'our', 'our', 'our', 'our'], indice 40
word: appointment retrieval:['preface', 'honour', 'milner', 'to', 'instance'], indice 43
word: woven retrieval:['womb', 'one', 'one', 'one', 'one'], indice 44
word: bustle retrieval:['loss', 'also', 'also', 'must', 'must'], indice 48
word: distinctively retrieval:['to', 'to', 'to', 'to', 'to'], indice 58
word: ampler retrieval:['man', 'man', 'man', 'man', 'man'], indice 59
word: pales retrieval:['held', 'held', 'held', 'held', 'avail'], indice 68
word: ancestral retrieval:['sunrise', 'ants', 'similar', 'grandparents', 'lamp'], indice 69
word: administrator retrieval:['treated', 'endured', 'e', 'penetrate', 'made'], indice 74
word: lantern retrieval:['him', 'him', 'him', 'him', 'him'], in

In [24]:
print(count)

430


In [25]:
count = 0
words_w2v_unique = []
for i in range(len(words_w2v)):
    word = words_w2v[i][0]
    if words_count[word] == 1:
        words_w2v_unique.append(words_w2v[i])
        count += 1
        print(f"word: {word} retrieval:{words_w2v[i][1]}, indice {i}")    

word: wielded retrieval:['sealed', 'will', 'will', 'will', 'will'], indice 3
word: performance retrieval:['reform', 'for', 'for', 'for', 'for'], indice 12
word: charwoman retrieval:['bun', 'them', 'reward', 'them', 'them'], indice 13
word: darker retrieval:['dark', 'dark', 'dark', 'dark', 'dark'], indice 19
word: appointment retrieval:['drainage', 'marriage', 'complained', 'bun', 'meantime'], indice 20
word: woven retrieval:['womb', 'wave', 'rudely', 'wave', 'ward'], indice 21
word: bustle retrieval:['bells', 'buffalo', 'robust', 'mall', 'solve'], indice 22
word: distinctively retrieval:['indistinct', 'existing', 'indistinctly', 'instinctive', 'league'], indice 26
word: ampler retrieval:['man', 'man', 'man', 'man', 'man'], indice 27
word: pales retrieval:['rows', 'of', 'of', 'of', 'of'], indice 30
word: ancestral retrieval:['senses', 'external', 'thence', 'mosaic', 'remonstrances'], indice 31
word: administrator retrieval:['witted', 'reduce', 'misuse', 'limit', 'mating'], indice 36
wor

In [26]:
words_w2v_counter = [item[0] for item in words_w2v_unique]
words_encodec_counter = [item[0] for item in words_encodec_unique]
words_encodec_mel_counter = [item[0] for item in words_encodec_mel_unique]

common_words = set(words_w2v_counter) & set(words_encodec_counter) & set(words_encodec_mel_counter)


data = {
    'word': list(common_words),
    'encodec Encodec': [next(item[1] for item in words_encodec_unique if item[0] == word) for word in common_words],
    'Encodec Mel': [next(item[1] for item in words_encodec_mel_unique if item[0] == word) for word in common_words],
    'w2v': [next(item[1] for item in words_w2v_unique if item[0] == word) for word in common_words]
}

df_unique = pd.DataFrame(data)

df_unique.to_csv('retrieval_unique.csv')


In [27]:
df_unique[:150]

word                                    encodec Encodec  \
0         sonnet                     [some, some, some, some, some]   
1      ancestral              [sunrise, known, founded, man, loves]   
2    strangeness                          [she, she, she, she, she]   
3             hi                                    [i, i, i, i, i]   
4          clams                  [wines, wines, wines, wine, wine]   
..           ...                                                ...   
145   background          [arms, unconsciously, asking, chat, arms]   
146   uproarious                          [are, are, are, are, are]   
147     ramparts                          [our, raw, our, our, our]   
148    frontiers  [philosophers, philosophers, reform, refer, re...   
149        friar                          [air, air, air, air, air]   

                                          Encodec Mel  \
0                           [son, son, son, son, son]   
1        [sunrise, ants, similar, grandparents, lamp]   
2    [sustains, strange, strange, strange, strangest]   
3                                     [i, i, i, i, i]   
4                   [wines, wines, wines, wine, wine]   
..                                                ...   
145             [ground, round, ground, round, round]   
146                         [are, are, are, are, are]   
147           [robber, robber, powers, powers, river]   
148                   [tears, far, tears, tears, far]   
149                    [fire, fire, fire, fire, fire]   

                                                   w2v  
0                  [amount, sauce, sauce, sunrise, on]  
1    [senses, external, thence, mosaic, remonstrances]  
2       [suspicion, suspicion, serge, assumes, secesh]  
3                                      [i, i, i, i, i]  
4                  [wines, wines, wines, class, cells]  
..                                                 ...  
145           [ground, ground, ground, ground, ground]  
146                [solve, rug, faults, fervor, logos]  
147                    [raw, ours, ours, ours, harbor]  
148        [armies, fifth, senses, teacher, fantasies]  
149                     [fire, fire, fire, fire, fire]  

[150 rows x 4 columns]

What happens with the other words?

In [28]:
words_encodec_not_unique = []
for i in range(len(words_encodec)):
    word = words_encodec[i][0]
    if not words_count[word] == 1:
        words_encodec_not_unique.append(words_encodec[i])
        print(f'word: {word}, retrieval: { words_encodec[i][1]}')

word: some, retrieval: ['insist', 'sooner', 'was', 'miss', 'was']
word: this, retrieval: ['the', 'the', 'the', 'the', 'the']
word: fire, retrieval: ['are', 'are', 'are', 'are', 'are']
word: never, retrieval: ['am', 'am', 'am', 'am', 'am']
word: excitement, retrieval: ['nile', 'wonder', 'autumn', 'general', 'twilight']
word: probably, retrieval: ['wildly', 'emperors', 'little', 'solved', 'little']
word: cross, retrieval: ['karl', 'karl', 'hot', 'crowd', 'casus']
word: and, retrieval: ['in', 'in', 'in', 'in', 'in']
word: no, retrieval: ['mode', 'mode', 'mode', 'mode', 'mode']
word: pocket, retrieval: ['back', 'back', 'up', 'up', 'up']
word: voice, retrieval: ['oil', 'war', 'tours', 'holgrave', 'will']
word: an, retrieval: ['in', 'in', 'in', 'in', 'in']
word: keep, retrieval: ['he', 'he', 'he', 'he', 'he']
word: recipe, retrieval: ['be', 'be', 'be', 'be', 'be']
word: that, retrieval: ['the', 'the', 'the', 'the', 'the']
word: for, retrieval: ['were', 'her', 'her', 'her', 'her']
word: durin

In [29]:
words_encodec_mel_not_unique = []
for i in range(len(words_encodec_mel)):
    word = words_encodec_mel[i][0]
    if not words_count[word] == 1:
        words_encodec_mel_not_unique.append(words_encodec_mel[i])
        print(f'word: {word}, retrieval: { words_encodec_mel[i][1]}')

word: this, retrieval: ['was', 'was', 'was', 'was', 'was']
word: fire, retrieval: ['are', 'are', 'are', 'are', 'are']
word: excitement, retrieval: ['it', 'it', 'it', 'it', 'it']
word: and, retrieval: ['in', 'in', 'in', 'in', 'in']
word: garden, retrieval: ['god', 'god', 'god', 'god', 'god']
word: no, retrieval: ['know', 'know', 'know', 'know', 'know']
word: pocket, retrieval: ['talk', 'talk', 'talk', 'talk', 'talk']
word: recipe, retrieval: ['be', 'be', 'be', 'be', 'be']
word: that, retrieval: ['the', 'the', 'the', 'the', 'the']
word: their, retrieval: ['are', 'are', 'are', 'are', 'are']
word: turban, retrieval: ['turn', 'turn', 'turn', 'turn', 'turnips']
word: tore, retrieval: ['wore', 'swore', 'poor', 'poor', 'poor']
word: office, retrieval: ['off', 'off', 'off', 'off', 'off']
word: drouet, retrieval: ['roof', 'roof', 'roof', 'roof', 'roof']
word: we, retrieval: ['he', 'he', 'he', 'he', 'he']
word: necessity, retrieval: ['see', 'see', 'see', 'see', 'see']
word: panada, retrieval: ['u

In [30]:
words_w2v_not_unique = []
for i in range(len(words_w2v)):
    word = words_w2v[i][0]
    if not words_count[word] == 1:
        words_w2v_not_unique.append(words_w2v[i])
        print(f'word: {word}, retrieval: { words_w2v[i][1]}')

word: cross, retrieval: ['across', 'across', 'across', 'across', 'across']
word: no, retrieval: ['now', 'now', 'now', 'now', 'now']
word: recipe, retrieval: ['stew', 'christie', 'fish', 'he', 'he']
word: drouet, retrieval: ['herbert', 'ardour', 'herbert', 'ardor', 'bread']
word: panada, retrieval: ['canada', 'add', 'atlanta', 'add', 'pleroma']
word: quarters, retrieval: ['quarter', 'quarter', 'quarter', 'corner', 'rope']
word: olden, retrieval: ['old', 'old', 'old', 'old', 'old']
word: call, retrieval: ['called', 'called', 'called', 'called', 'called']
word: venus, retrieval: ['v', 'v', 'main', 'main', 'phoebe']
word: while, retrieval: ['wild', 'wild', 'wild', 'wild', 'wild']
word: edition, retrieval: ['tuition', 'gifted', 'to', 'to', 'distinctions']
word: nor, retrieval: ['or', 'or', 'or', 'or', 'or']
word: whisper, retrieval: ['fervor', 'ardor', 'sward', 'collar', 'ardour']
word: beaten, retrieval: ['beat', 'beat', 'beat', 'beat', 'beat']
word: whom, retrieval: ['who', 'who', 'who', 

In [31]:
words_w2v_counter = [item[0] for item in words_w2v_not_unique]
words_encodec_counter = [item[0] for item in words_encodec_not_unique]
words_encodec_mel_counter = [item[0] for item in words_encodec_mel_not_unique]


common_words = set(words_w2v_counter) & set(words_encodec_counter) & set(words_encodec_mel_counter)

data = {
    'word': list(common_words),
    'encodec ec': [next(item[1] for item in words_encodec_not_unique if item[0] == word) for word in common_words],
    'encodec mel': [next(item[1] for item in words_encodec_mel_not_unique if item[0] == word) for word in common_words],
    'w2v': [next(item[1] for item in words_w2v_not_unique if item[0] == word) for word in common_words]
}

df_not_unique = pd.DataFrame(data)

df_not_unique.to_csv('retrieval_not_unique.csv')

In [32]:
df_not_unique[:400]

word                                 encodec ec  \
0         inches                  [she, she, she, she, she]   
1    distinction           [in, should, should, should, in]   
2          jesus                  [she, she, she, she, she]   
3         orders                       [or, or, or, or, or]   
4       captured   [should, should, should, should, should]   
..           ...                                        ...   
395       hansel            [sin, sail, hid, handed, hands]   
396         ones         [level, honour, one, willing, uns]   
397      witness                       [in, in, in, in, in]   
398   democratic  [captured, problem, cannon, picket, mark]   
399    therefore               [are, are, were, were, were]   

                                       encodec mel  \
0                             [as, as, as, as, as]   
1    [ancient, ancient, ancient, ancient, nations]   
2                          [is, is, is, is, these]   
3              [doors, doors, doors, doors, doors]   
4                   [half, half, half, half, half]   
..                                             ...   
395            [hands, hands, hands, hands, hands]   
396                   [uns, them, one, them, them]   
397              [assassin, semblance, at, at, at]   
398               [trap, trap, trap, crept, cramp]   
399                  [four, four, four, for, four]   

                                                w2v  
0                         [his, his, his, his, his]  
1     [student, string, contingent, kings, streams]  
2             [choose, choose, these, sees, phases]  
3               [order, order, order, order, order]  
4               [after, after, after, after, after]  
..                                              ...  
395             [somehow, penalties, has, thy, has]  
396                       [one, one, one, one, one]  
397                  [this, this, this, this, this]  
398  [sack, credit, aristocratic, comrade, comrade]  
399                      [for, rear, for, for, for]  

[400 rows x 4 columns]

In [33]:
words_w2v_counter = [item[0] for item in words_w2v]
words_encodec_counter = [item[0] for item in words_encodec]
words_encodec_mel_counter = [item[0] for item in words_encodec_mel]

w2v_words = set(words_w2v_counter) - set(words_encodec_counter) - set(words_encodec_mel_counter)

data = {
    'word': list(w2v_words),
    'w2v': [next(item[1] for item in words_w2v if item[0] == word) for word in w2v_words]
}

df_unique_w2v = pd.DataFrame(data)

df_unique_w2v.to_csv('retrieval_unique_w2v.csv')


In [34]:
words_w2v_counter = [item[0] for item in words_w2v]
words_encodec_counter = [item[0] for item in words_encodec]
words_encodec_mel_counter = [item[0] for item in words_encodec_mel]

encodec_words =  set(words_encodec_counter) - set(words_w2v_counter) - set(words_encodec_mel_counter)

data = {
    'word': list(encodec_words),
    'encodec': [next(item[1] for item in words_encodec if item[0] == word) for word in encodec_words]
}

df_unique_encodec = pd.DataFrame(data)

df_unique_encodec.to_csv('retrieval_unique_encodec_ec.csv')

In [35]:
words_w2v_counter = [item[0] for item in words_w2v]
words_encodec_counter = [item[0] for item in words_encodec]
words_encodec_mel_counter = [item[0] for item in words_encodec_mel]

encodec_words_mel =  set(words_encodec_mel_counter) - set(words_w2v_counter) - set(words_encodec_counter)

data = {
    'word': list(encodec_words_mel),
    'encodec': [next(item[1] for item in words_encodec_mel if item[0] == word) for word in encodec_words_mel]
}

df_unique_encodec_mel = pd.DataFrame(data)

df_unique_encodec_mel.to_csv('retrieval_unique_encodec_mel.csv')

In [36]:
words_w2v_counter = [item[0] for item in words_w2v]
words_encodec_counter = [item[0] for item in words_encodec]
words_encodec_mel_counter = [item[0] for item in words_encodec_mel]


encodec_words_mel =  set(words_encodec_mel_counter) - set(words_w2v_counter)

data = {
    'word': list(encodec_words_mel),
    'encodec MEl': [next(item[1] for item in words_encodec_mel if item[0] == word) for word in encodec_words_mel]
}

df_unique_encodec_mel_notw2v = pd.DataFrame(data)

df_unique_encodec_mel_notw2v[250:330]

word                            encodec MEl
250     hold              [old, old, old, old, old]
251  sending          [setting, seen, in, in, seen]
252      gun            [men, men, men, done, mens]
253      tin              [man, man, man, man, man]
254     otto                        [i, i, i, i, i]
..       ...                                    ...
325     soul  [loyal, loyally, whole, whole, whole]
326   showed         [show, show, show, show, show]
327  brother    [other, other, other, other, other]
328     less                   [as, as, as, as, as]
329    honor              [all, all, all, all, all]

[80 rows x 2 columns]